spec/classes/plugin_ansible_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+describe 'pulpcore::plugin::ansible' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+      let(:pre_condition) { 'include pulpcore' }
+
+      it { is_expected.to compile.with_all_deps }
+      it { is_expected.to contain_package('python3-pulp-ansible') }
+      it { is_expected.to contain_pulpcore__plugin('ansible') }
+      it do
+        is_expected.to compile.with_all_deps
+        is_expected.to contain_pulpcore__plugin('ansible')
+          .that_subscribes_to('Class[Pulpcore::Install]')
+          .that_notifies(['Class[Pulpcore::Database]', 'Class[Pulpcore::Service]'])
+        is_expected.to contain_apache__vhost__fragment('pulpcore-https-plugin-ansible')
+                           .with_content(
+<<CONTENT
+
+  ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
+  ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
+CONTENT
+                           )
```

Comment:
Is there any way we can make this nicer to look at ?

I tried:

```
          .with_content(
            <<~CONTENT.chomp
            
              \s\sProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
              \s\sProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
            CONTENT
          )
```

But rpsec doesn't seem to be able to treat the `\s` sequences as spaces:

```
       expected that the catalogue would contain Apache::Vhost::Fragment[pulpcore-https-plugin-ansible] with content set to supplied string
       Diff:
         <The diff is empty, are your objects producing identical `#inspect` output?>
```

The `#strip_heredoc` method from rails would do what I'm looking for but it's not available here. I also tried interpolation to get the two spaces but rspec fails in the same way as above.

What finally worked for me is defining the `VHOST_CONTENT` at the top of the file:

```
VHOST_CONTENT=<<HEREDOC
  ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
  ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
HEREDOC
```

And then the test becomes:

```suggestion
          .with_content(VHOST_CONTENT)
```

Open to other ideas if you have them :)

Reply:
> But rpsec doesn't seem to be able to treat the `\s` sequences as spaces:

That's because heredoc is a string, not a regex.

> Open to other ideas if you have them :)

You can consider a let block which is probably better than a constant:
```ruby
let(:expected_vhost_content) do
  'the expected content'
end
```

Another option is to use `.with_content` twice if you don't care where it exactly is, just that it's there:
```ruby
.with_content(%r{ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock\|http://pulpcore-api/pulp_ansible/galaxy/})
.with_content(%{ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock\|http://pulpcore-api/pulp_ansible/galaxy/})

Reply:
I tried moving it to a let as a string:

```
      let(:expected_vhost_content) do
        "\n  ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/\n" +
            "  ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/"
      end
```

```
        is_expected.to contain_apache__vhost__fragment('pulpcore-https-plugin-ansible')
                           .with_content(expected_vhost_content)
```

But i get:
```

  1) pulpcore::plugin::ansible on redhat-7-x86_64 is expected to contain Apache::Vhost::Fragment[pulpcore-https-plugin-ansible] with content  supplied string
     Failure/Error:
       is_expected.to contain_apache__vhost__fragment('pulpcore-https-plugin-ansible')
                          .with_content(expected_vhost_content)
       expected that the catalogue would contain Apache::Vhost::Fragment[pulpcore-https-plugin-ansible] with content set to supplied string
       Diff:
         <The diff is empty, are your objects producing identical `#inspect` output?>
```

but defining the let with a HEREDOC works:
```
      let(:expected_vhost_content) do
<<CONTENT

  ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
  ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
CONTENT
      end
```

IDK i'm fine with the heredoc, in this case i think its important to verify the content exactly (and not in a location)

Reply:
I feel like a large inline block of texts makes it harder to read and would prefer a `let` block.

Reply:
to clarify, you mean  a let block with heredoc as well?

Reply:
went ahead and pushed that, let me know if that wasn't what you meant